src/character/mod.rs

```diff
@@ -152,6 +161,14 @@ impl Character {
             damage / (1 + self.level - receiver.level)
         }
     }
+
+    pub fn produces_condition(&self, receiver: &Self) -> bool {
+        receiver.condition.is_none()
+    }
+
+    pub fn receive_condition(&mut self, condition: Option<Condition>) {
+        self.condition = condition;
```

Comment:
if you are using options, you should make the Some(condition) call inside of this function instead of having the caller know that it's stored as an option